bc-backend/src/main/java/org/jboss/da/bc/impl/ProjectHiearchyCreator.java

```diff
@@ -154,7 +168,7 @@ private void setDependencies(ProjectHiearchy hiearchy) {
     }
 
     private Set<ProjectHiearchy> toProjectHiearchies(Collection<GAV> gavs) {
-        return gavs.parallelStream().map(dep -> toProjectHiearchy(dep)).collect(Collectors.toSet());
+        return gavs.stream().map(dep -> toProjectHiearchy(dep)).collect(Collectors.toSet());
```

Comment:
toProjectHiearchy starts asynchronous tasks and adds them to the ``futures`` field. It then waits for them in the entry (public) methods.